File: python/anagram/anagram.py
def detect_anagrams(txt, candidates):
    return [c for c in candidates if _is_anagram(c, txt)]

def _is_anagram(a, b):
    return (
        a.lower() != b.lower() and
        _sort(a) == _sort(b)
    )

def _sort(txt):
    return ''.join(sorted(txt.lower()))

File: javascript/anagram/anagram.js
const sort = word => [...word].sort().join('');

class Anagram {

    constructor(word) {
        this.word = word.toLowerCase();
        this.sorted = sort(this.word);
    }

    matches(...input) {
        input = (Array.isArray(input[0])) ? input[0] : input;
        return input.filter(word => {
            let lower = word.toLowerCase();
            return (lower !== this.word && sort(lower) === this.sorted);
        });
    }
}

module.exports = Anagram;